ui/src/app/machines/components/HeaderStrip/AddHardwareMenu/AddHardwareMenu.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+
+import { general as generalActions } from "app/base/actions";
+import { general as generalSelectors } from "app/base/selectors";
+import ContextualMenu from "app/base/components/ContextualMenu";
+
+const getAddHardwareLinks = navigationOptions => {
+  const links = [
+    {
+      children: "Machine",
+      element: Link,
+      to: "/machines/add"
+    },
+    {
+      children: "Chassis",
+      element: Link,
+      to: "/machines/chassis/add"
+    }
+  ];
+
+  return navigationOptions.rsd
+    ? links.concat({
+        children: "RSD",
+        element: "a",
+        href: `${process.env.REACT_APP_ANGULAR_BASENAME}/rsd`
+      })
+    : links;
+};
+
+export const AddHardwareMenu = () => {
+  const dispatch = useDispatch();
+  const navigationOptions = useSelector(generalSelectors.navigationOptions.get);
+  const navigationOptionsLoaded = useSelector(
+    generalSelectors.navigationOptions.loaded
+  );
+
+  useEffect(() => {
+    if (!navigationOptionsLoaded) {
```

Comment:
Was this fetching the navigation options even if they existed in the store? I was under the impression that the saga handled checking if they were loaded. If not this might be a bug.

Reply:
Oh you're 100% right. I thought it was refetching every time but it's just calling the action "stem" (`FETCH_MACHINE` not `FETCH_MACHINE_START`). I'll remove the loaded checks.